.github/workflows/test-ci.yaml

```diff
@@ -0,0 +1,228 @@
+name: test-CI
+
+concurrency:
+  group: ci-${{ github.run_id }}
+  cancel-in-progress: true
+
+on:
+  issue_comment:
```

Comment:
We also wants to run tests on push too. One way is to extract the test jobs into a separate reusable workflow: https://docs.github.com/en/actions/using-workflows/reusing-workflows

Reply:
I thought since the push and issue_comment events are so different, it would be simpler to just make a new workflow for push - but if necessary, I can make the test jobs into a reusable workflow I think?

Reply:
Yehh right! We should split the test jobs into resusable workflow!

Reply:
And have 2 workflows: push + issue_comment that calls it.